src/sets.jl

```diff
@@ -263,6 +349,16 @@ end
 dual_set(s::RotatedSecondOrderCone) = copy(s)
 dual_set_type(::Type{RotatedSecondOrderCone}) = RotatedSecondOrderCone
 
+function distance_to_set(::DefaultDistance, v::AbstractVector{<:Real}, s::RotatedSecondOrderCone)
+    _check_dimension(v, s)
+    t = v[1]
+    u = v[2]
+    xs = v[3:end]
+    return LinearAlgebra.norm2(
+        (max(-t, zero(t)), max(-u, zero(u)), max(dot(xs,xs) - 2 * t * u))
```

Comment:
The core issue here is that the cone is characterized by three constraints.
With vector-distance, this would have been:
```
max(-u, 0)
max(-v, 0)
max(-2tu + norm_squared(x) , 0)
```

since we don't have that, the l2-norm of these three epigraph violations is the thing computed here.